Fix _get_pressed() returning list of booleans

vaibhav-sangwan opened the pull request:
The changes commited in e4d3122 converted the type of self.__keystate from bool array into dict(int: bool). However, _get_pressed() still returns a list of 134 booleans. Checking key presses for a key such as pygame.K_RIGHT(1073741903) would go past the size of the list resulting in an IndexError.

* return self.__keystate dict from _get_pressed() instead of it's values.

Solves the issue encountered in https://github.com/sugarlabs/ball-and-brick-activity/pull/24#issuecomment-2053740586.

@quozl, @walterbender, @chimosky please review.

quozl commented:
Well done!

chimosky commented:
What activity did you notice this in? Any steps to reproduce?

vaibhav-sangwan commented:
The issue was originally encountered in [BallAndBrick activity](https://github.com/sugarlabs/ball-and-brick-activity/pull/24#issuecomment-2053740586). My testing environment consists of sugar 0.118 on Ubuntu 22.04 (Pygame 2.5.2, Python 3.10.12).
To reproduce the issue, you can test [af89d7c](https://github.com/sugarlabs/ball-and-brick-activity/pull/24/commits/af89d7cc50209a946c4a1d2062a29f853a5012d6). On the landing screen, press C to hop into the game, and then pressing the right and left arrow keys won't trigger any movement. The following error is shown in logs - 
```
Traceback (most recent call last):
 File "/home/vishu/Activities/ball-and-brick-activity/game.py", line 601, in run
   settings()
 File "/home/vishu/Activities/ball-and-brick-activity/game.py", line 471, in settings
   gameLoop()
 File "/home/vishu/Activities/ball-and-brick-activity/game.py", line 512, in gameLoop
   check_input()
 File "/home/vishu/Activities/ball-and-brick-activity/game.py", line 209, in check_input
   if keys[pygame.K_LEFT]:
IndexError: list index out of range
```

chimosky commented:
I tested the changes you made and it works just fine without this change to sugargame.

vaibhav-sangwan commented:
Did you test all the commits? The changes made in [b03de8c ](https://github.com/sugarlabs/ball-and-brick-activity/commit/b03de8c22e7aff38f33ddb4304623ee6249f3150) replaced the usage of pygame.key.get_pressed() and thus the BallAndBrick activity would work fine without any changes to sugargame.
However, while testing the activity after commit [9eb67d3](https://github.com/sugarlabs/ball-and-brick-activity/commit/9eb67d39d7361a3886563cbedccde7ef7577b2c7), the activity froze on welcome page and in the logs, the above mentioned error was shown.

vaibhav-sangwan committed:
Fix _get_pressed() returning list of booleans

_get_pressed() returns a list of 134 booleans. Checking key presses for a key
such as pygame.K_RIGHT(1073741903) would go past the 134 element list resulting
in an IndexError.

* return self.__keystate dict from _get_pressed() instead of it's values.

vaibhav-sangwan commented:
I wanted to change the commit message only but mistakenly pushed some unwanted changes. So had to force push twice.
